backend/oasst_backend/user_stats_repository.py

```diff
@@ -226,35 +226,35 @@ def update_ranks(self, time_frame: UserStatsTimeFrame = None):
         Update user_stats ranks. The persisted rank values allow to
         quickly the rank of a single user and to query nearby users.
         """
-
-        # todo: convert sql to sqlalchemy query..
-        # ranks = self.session.query(
-        #     func.row_number()
-        #     .over(partition_by=UserStats.time_frame, order_by=[UserStats.leader_score.desc(), UserStats.user_id])
-        #     .label("rank"),
-        #     UserStats.user_id,
-        #     UserStats.time_frame,
-        # )
-
-        sql_update_rank = """
--- update rank
-UPDATE user_stats us
-SET "rank" = r."rank"
-FROM
-    (SELECT
-        ROW_NUMBER () OVER(
-            PARTITION BY time_frame
-            ORDER BY leader_score DESC, user_id
-        ) AS "rank", user_id, time_frame
-    FROM user_stats
-    WHERE (:time_frame IS NULL OR time_frame = :time_frame)) AS r
-WHERE
-    us.user_id = r.user_id
-    AND us.time_frame = r.time_frame;"""
-        r = self.session.execute(
-            text(sql_update_rank), {"time_frame": time_frame.value if time_frame is not None else None}
-        )
-        logger.debug(f"pre_compute_ranks updated({time_frame=}) {r.rowcount} rows.")
+        try:
+            # sql alchemy implementation
+            subquery = (
+                select(
+                    [
+                        func.row_number()
+                        .over(
+                            partition_by=UserStats.time_frame,
+                            order_by=[UserStats.leader_score.desc(), UserStats.user_id],
+                        )
+                        .label("rank"),
+                        UserStats.user_id,
+                        UserStats.time_frame,
+                    ]
+                )
+                .where(UserStats.time_frame == time_frame.value if time_frame is not None else None)
```

Comment:
Did you verify (e.g. by printing the query to inspect the generated sql) that `time_frame.value if time_frame is not None else None` is not making it into the SQL? .. otherwise I would recommend to build the query dynamically based on the value of `time_frame`.